iaso/api/payments/serializers.py

```diff
@@ -76,28 +83,50 @@ class Meta:
 class PaymentLotSerializer(serializers.ModelSerializer):
     payments = serializers.SerializerMethodField()
     task = NestedTaskSerializer(read_only=True)
+    can_see_change_requests = serializers.SerializerMethodField()
 
     class Meta:
         model = PaymentLot
-        fields = ["id", "name", "status", "created_at", "created_by", "payments", "comment", "task"]
+        fields = [
+            "id",
+            "name",
+            "status",
+            "created_at",
+            "created_by",
+            "payments",
+            "comment",
+            "task",
+            "can_see_change_requests",
+        ]
         read_only_fields = ["id", "created_at"]
 
     pagination_class = PaymentPagination
     created_by = UserNestedSerializer()
     created_at = TimestampField(read_only=True)
 
+    def get_can_see_change_requests(self, obj):
+        user = self.context.get("request").user
+        if user.is_superuser:
+            return True
+        user_org_units = sorted(list(user.iaso_profile.get_hierarchy_for_user().values_list("id", flat=True)))
+        change_requests_org_units_for_lot = sorted(
+            list(OrgUnitChangeRequest.objects.filter(payment__in=obj.payments.all()).values_list("org_unit__id"))
+        )
+        return user_org_units == change_requests_org_units_for_lot
```

Comment:
user_org_units = set(user.iaso_profile.get_hierarchy_for_user().values_list("id", flat=True))
        change_requests_org_units_for_lot = (
            OrgUnitChangeRequest.objects.filter(payment__in=obj.payments.all())
            .values_list("org_unit__id", flat=True)
            .distinct()
        )
        return set(change_requests_org_units_for_lot).issubset(user_org_units)
```suggestion
        user_org_units = sorted(list(user.iaso_profile.get_hierarchy_for_user().values_list("id", flat=True)))
        change_requests_org_units_for_lot = sorted(
            list(OrgUnitChangeRequest.objects.filter(payment__in=obj.payments.all()).values_list("org_unit__id"))
        )
        return user_org_units == change_requests_org_units_for_lot
```